src/builders/indexable-post-builder.php

```diff
@@ -265,8 +276,6 @@ protected function has_public_posts( $indexable ) {
 	 * @return bool|null True for noindex, false for index, null for default of parent/type.
 	 */
 	protected function get_robots_noindex( $value ) {
-		$value = (int) $value;
```

Comment:
Why is this being removed ? (or rather moved)

Reply:
Because it didn't match the declared type in phpdoc. Now, I know that it's not the same as proper type declarations, but since this method is protected and it's only used here, I wanted to adhere to its type and move the type cast.
But I get your point, i'll restore the type cast in the method and keep the cast at the caller too, so when it's time to use declarations we only need change this once place.

Reply:
As things were, the `(int)` cast was providing a similar form of type safety as a type declaration would. A scalar `int` type declaration will juggle all scalars to the declared type and will error on non-scalars (`null`, array, object, resource).
Until a type declaration can be added, the `(int)` declaration will do the same with the one different being the handling of `null`. (the cast handles it, the type declaration would not)

Reply:
(in other words, as long as what the caller provides is scalar, no type cast is needed there)